crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -223,34 +225,60 @@ impl SettlementEncoder {
         // <=> order.sellAmount.mul(sellPrice)  >= order.buyAmount.mul(sell_price * order.sellAmount / order.buyAmount)
         // <=> order.sellAmount.mul(sellPrice)  >= order.buyAmount.mul(buyPrice)
         // <=> equation from smart contract
-
-        let buy_price = self
-            .clearing_prices
+        //
+        self.clearing_prices
             .get(&order.data.sell_token)
-            .context("settlement missing sell token")?
+            .unwrap_or(&order.data.buy_amount)
```

Comment:
Because `add_custom_price_trade()` either takes the existing sell token price or adds the new one we have to follow the same process here as well.